My i18n

shibainurou opened the pull request.

shibainurou committed:
レイアウト変更

shibainurou committed:
i18n対応

yoshitsugu commented on app/views/books/_book.html.erb:
```diff
@@ -1,21 +1,22 @@
   <p>
-    <strong>Title:</strong>
+    <strong><%= t("title") %>:</strong>
+    <strong>form :</strong>
```
この `form:` ってなんでしょう？ デバッグ用のものがのこってますかね？

shibainurou replied:
不要なコードが残っていました。削除します。

yoshitsugu commented on app/views/books/_book.html.erb:
```diff
@@ -1,21 +1,22 @@
 
   <p>
-    <strong>Memo:</strong>
+    <strong><%= t("memo") %>:</strong>
```
属性名は `human_attribute_name` を使うとよいです。

https://railsguides.jp/i18n.html#active-record%E3%83%A2%E3%83%87%E3%83%AB%E3%81%A7%E7%BF%BB%E8%A8%B3%E3%82%92%E8%A1%8C%E3%81%AA%E3%81%86

フレームワークなどで提供されている便利メソッドなどはできるだけ使うようにすると以下のメリットがあります。
- フレームワーク側で(速度などの面で)最適化される可能性がある
- 見落としがちな落とし穴を避けた、安全なプログラムになりやすい
- 制約が増えるので、バグを発見しやすい

shibainurou replied:
調査不足でした。human_attribute_nameを使うように修正しました。

yoshitsugu commented on config/locales/en.yml:
```diff
@@ -30,4 +30,34 @@
+  new:
+    title: "New book"
+    message:
+      finish: "Book was successfully created."
```
flashメッセージ用の訳ですが、もし今後おなじようなcontrollerが増えていったとして、都度 `XXX was successfully created` というメッセージを追加するのは手間ですよね。 `XXX` の部分を変数としてわたせるようにしてみてください。

https://railsguides.jp/i18n.html#%E8%A8%B3%E6%96%87%E3%81%AB%E5%A4%89%E6%95%B0%E3%82%92%E6%B8%A1%E3%81%99

また、モデル名は `model_name.human` を使うとよいかと思います。

https://railsguides.jp/i18n.html#active-record%E3%83%A2%E3%83%87%E3%83%AB%E3%81%A7%E7%BF%BB%E8%A8%B3%E3%82%92%E8%A1%8C%E3%81%AA%E3%81%86

shibainurou replied:
こちらも調査不足でした。
変数を使うように修正しました。また変数はモデル名を使うように修正しました。

yoshitsugu reviewed (commented):
コメントしました。

shibainurou committed:
i18nの修正
”本”を含む文章をパラメータを使うように修正
文字列"本"はBook.model_name.humanを使うように修正
英語のbookは単数と複数を使い分けるように修正

shibainurou committed:
不要なコードの削除

yoshitsugu reviewed (approved):
OKです〜
